otx/algorithms/segmentation/tasks/inference.py

```diff
@@ -217,11 +217,10 @@ def _init_recipe(self):
         logger.info(f"initialized recipe = {recipe}")
 
     def _update_stage_module(self, stage_module: str):
-        if self.train_type == TrainType.SEMISUPERVISED:
-            if stage_module == "SegTrainer":
-                return "SemiSegTrainer"
-            if stage_module == "SegInferrer":
-                return "SemiSegInferrer"
+        module_prefix = {TrainType.SEMISUPERVISED: "Semi", TrainType.INCREMENTAL: "Incr"}
```

Comment:
It'd be better if we have a `SegTask` that has 2 methods - `train` and `inference`, so we don't need 2 more classes (`xTrainer` and `xInferrer` for every task). For seg, we have `SegTrainer`, `SegInferrer`, SemiSegTrainer, SemiSegInferrer, IncrSegTrainer, and IncrSegInferrer, that's overkill. Instead, we can have `SegTask`, `SemiSegTask`, `IncrementalSegTask`. From 6 to 3. 

I also don't think this a good way to get class using string, it's easily breakable, and the intention is unclear. Could we not tell which task class we are training? If parsed the task type, and the exact task CLI from `train.py`. Could we not get the class directly from that 2 pieces of information?

Reply:
I hope you refer to my comment above as well. And about module name, I will revise it to get the value with full name.